index.js

```diff
@@ -112,13 +112,19 @@ module.exports = {
       try {
         // update global doc with random number to compare it with the next req
         // see expressMiddleware in this file
-        const i18nGeneration = self.apos.utils.generateId();
+        const localeI18nGeneration = self.apos.utils.generateId();
         const query = { type: 'apostrophe-global' };
 
         if (self.apos.modules['apostrophe-workflow']) {
           query.workflowLocale = { $in: [piece.lang, piece.lang + '-draft'] };
         }
-        await self.apos.docs.db.update(query, { $set: { i18nGeneration } }, { multi: true });
+        await self.apos.docs.db.updateMany(query, {
+          $set: {
+            i18nGeneration: {
```

Comment:
This would replace the entire i18nGeneration object including other sub-objects, but also it should not be necessary, see my comment above that the query being matched is already locale specific.